docs(howto): rescue your data MTA-1489 (#124)
scaleway/docs-content: dedibox/dedicated-servers/how-to/rescue-your-data.mdx

---
title: How to rescue your data
description: Learn how to download backups of your data using rescue mode.
---

<Message type="requirement">

  - You have an account and are logged into the [Dedibox Console](https://console.online.net)
  - You have [created](/dedibox/dedicated-servers/how-to/order-a-server) and [installed](/dedibox/dedicated-servers/how-to/install-a-server) a dedicated server

</Message>

Whatever your operating system (Windows or Linux), if your server does not have a hardware failure, you can recover your files via the rescue mode.

## How to boot into rescue mode

> Start your server in rescue mode (emergency system) via your [Dedibox console](/ddedibox/dedicated-servers/how-to/use-rescue-mode).

Reviewer: ```suggestion
From your [Dedibox console](/dedibox/dedicated-servers/how-to/use-rescue-mode), start your server in rescue mode (emergency system) and wait until your machine is available. This operation can take up to 15 minutes depending on the server's hardware
```

> Wait until your machine has become available (it takes 5 to 15 minutes depending on the servers hardware)

Reviewer: delete

Log into your server:
* Either using [PuttY](http://the.earth.li/~sgtatham/putty/latest/x86/putty.exe) on computers running [Windows](https://www.scaleway.com/en/docs/dedibox/dedicated-servers/how-to/connect-to-server/#from-windows)
* Or using the `ssh` command directly from a terminal on computers runnig [Linux or macOS](/dedibox/dedicated-servers/how-to/connect-to-server/#from-osx-(mac)-and-linux).

<Message type="tip">

The credentials (username and password) to log into the rescue system are displayed in your Dedibox console. 

</Message>

## How to gain root rights

> You are connected to your Dedibox via a regular user account, therefore you do not have rights to execute all required commands.

Reviewer 2: ```suggestion
You are now connected to your Dedibox in rescue mode via a regular user account, therefore you do not have rights to execute all required commands.
```

> This is the philosophy of Ubuntu, S root account to get the right to do what we want.

Reviewer 2: ```suggestion
This is the philosophy of Ubuntu, it is only with the root account that we get full rights to do what we want.
```

> This is the philosophy of Ubuntu, S root account to get the right to do what we want.

Reviewer: on supprime cette phrase non ?

<Message type="important">

> Changing into the root user implies that you can perform unsafe practices such as deleting files.

Reviewer 2: ```suggestion
Changing into the root user implies that you can also perform unsafe practices such as deleting files.
```

> Changing into the root user implies that you can perform unsafe practices such as deleting files.

Reviewer: ```suggestion
Changing into the root user implies that you can perform permanent commands such as deleting files.
```

> Perform these commands your own risk.

Reviewer: celle-ci aussi on peut supprimer

> Perform these commands your own risk.

Reviewer 2: ```suggestion
Perform these commands at your own risk.
```

</Message>

1. Switch into the `root` account by typing the following command, then press enter:

  ```
  sudo su
  ```

2. Enter the password of the rescue user, displayed in the console and confirm by pressing enter. 

  <Message type="tip">

  For security reasons, the password will not be shown when you type it.

  </Message>

## How to mount the partitions

To have access to your data using the rescue moude, you have to mount your partitions.

> If this step passes, then you are almost sure to be able to recover your files.

Reviewer 2: ```suggestion
If this step succeeds, then you are almost sure to be able to recover your files.
```

> Mount your partitions using the script available by typing, as `root` user, the command `mountall.sh`.  

Reviewer 2: ```suggestion
Mount your partitions by entering, as `root` user, the command `mountall.sh`.  
```

The partitions are then mounted in the `/mnt` directory

<Message type="note">

> You will notice some errors, the swap cannot be mounted, and many partitions do not exist. This is normal and you do not have to worry about it.

Reviewer 2: ```suggestion
You will notice some errors such as the swap cannot be mounted, and many partitions do not exist. This is normal and not a cause for concern.
```

> You will notice some errors, the swap cannot be mounted, and many partitions do not exist. This is normal and you do not have to worry about it.

Reviewer: ```suggestion
Some errors such as `the swap cannot be mounted` or `many partitions do not exist`. You can safely ignore these errors. 
```

</Message> 

## How to recover your data

> Depending on your level of knowledge, you can select your files, or even delete everything, you do what you want.

Reviewer 2: ```suggestion
Depending on your level of knowledge and your requirements, you can select the files you want to recover, delete files or whatever you need.
```

> Depending on your level of knowledge, you can select your files, or even delete everything, you do what you want.

Reviewer: on peut supprimer ca non ?

###  How to resuce your data on Dedibackup

> Here we will show you how to retrieve ALL the contents of your disk (including all partitions) on your [Dedibackup]() storage space.

Reviewer 2: missing link?

> Here we will show you how to retrieve ALL the contents of your disk (including all partitions) on your [Dedibackup]() storage space.

Reviewer: ```suggestion
The following procedure shows how to retrieve all the contents of your disk (including all partitions) on your [Dedibackup]() storage space.
```

> Here we will show you how to retrieve ALL the contents of your disk (including all partitions) on your [Dedibackup]() storage space.

Author: ```suggestion
Here we will show you how to retrieve ALL the contents of your disk (including all partitions) on your [Dedibackup](https://www.scaleway.com/en/dedibox/storage/dedibackup/) storage space.
```

> 1. Install an FTP client that will work by itself, and `screen` that allows it run it without your computer having to stay on with putty open.

Reviewer 2: ```suggestion
1. Install an FTP client and `screen` via the following command. `screen` allows the FTP client to run without your computer having to stay on with Putty open.
```

  ```
  apt -y install screen lftp
  ```

2. Launch `screen` bt typing the following command. Then press enter to launch the application:

  ```
  screen
  ```

3. Lauch the transfer of your data by running the following command:

  ```
  lftp -u YOUR_DEDIBACKUP_LOGIN,YOUR_DEDIBACKUP_PASSWORD dedibackup.online.net -e "mirror --reverse /mnt /; exit"
  ```

  * `YOUR_DEDIBACKUP_LOGIN` has to be replaced with your dedibackup login (typically sd- your server number).
  * `YOUR_DEDIBACKUP_PASSWORD` has to be replaced with your dedibackup password.

  <Message type="tip">

>   If you do not know the password, change it in the console in order to know it.

Reviewer 2: ```suggestion
  If you do not know the password, change it in the console to get a new password.
```

  </Message>

> Now, if all went well, all your files should be copying to your dedibackup space.

Reviewer: delete cette phrase

> Now, if all went well, all your files should be copying to your dedibackup space.

Author: ```suggestion
```

> You will be able to browse and retrieve them with any ftp client by connecting to dedibackup via an FTP client.

Reviewer 2: ```suggestion
You will be able to browse and retrieve them by connecting an FTP client to dedibackup.
```

> You will be able to browse and retrieve them with any ftp client by connecting to dedibackup via an FTP client.

Reviewer: ```suggestion
You should now be able to browse and retrieve all your files with any ftp client by connecting to dedibackup via an FTP client.
```

### Hox to backup to your personal PC (via SFTP)

> The recovery mode supports by default the connection using SSH, it is therefore possible to connect from your computer using a SFTP client to recover your files.

Reviewer 2: ```suggestion
The recovery mode supports connection using SSH by default. It is therefore possible to connect from your computer using a SFTP client to recover your files.
```

In this example we use the client [FileZilla](en/web/web-hosting/tutorials/ftp/configure-filezilla)

> To connect to the server, you are required to fill the follwoing values:

Reviewer 2: ```suggestion
To connect to the server, you are required to complete the following values:
```

> *  **Host**: The servers IP address

Reviewer 2: ```suggestion
*  **Host**: The server's IP address
```

*  **Identifier**: The user name used to connect to the server
*  **Password**: The password of the recovery system
*  **Port**: Port 22

<Message type="tip">

These details are available in your console once the recovery mode is launched.

</Message>

##### Configuration example

<Lightbox src="sftp.png"  alt="" /> 

Your data is located in the folders `/mnt`, and `/sda1`, or `/sda`.

> You can download your data before a reinstallation or the transfer on another server.

Reviewer 2: ```suggestion
You can download your data before you reinstall your server, or before transferring data to another server.
```

<Message type="tip">

> If you do not feel able / do not have time to do these manipulations, you can contact a [certified outsourcer](https://www.scaleway.com/en/dedibox/outsourcing/) to perform these tasks for you.

Reviewer 2: ```suggestion
If for any reason you are unable to follow this process, you can contact a [certified outsourcer](https://www.scaleway.com/en/dedibox/outsourcing/) to rescue your data for you.
```

   * [PuttY](http://the.earth.li/~sgtatham/putty/latest/x86/putty.exe) on computers running [Windows](https://www.scaleway.com/en/docs/dedibox/dedicated-servers/how-to/connect-to-server/#from-windows)
   * `ssh` directly from a terminal on computers runnig [Linux or macOS](/dedibox/dedicated-servers/how-to/connect-to-server/#from-osx-(mac)-and-linux).

<Message type="tip">

The credentials (username and password) to log into the rescue system are displayed in your Dedibox console. 

</Message>

## How to gain root rights

You are now connected to your Dedibox in rescue mode via a regular user account, therefore you do not have rights to execute all required commands.
This is the philosophy of Ubuntu, it is only with the root account that we get full rights to do what we want.

1. Switch into the `root` account by typing the following command, then press enter:

  ```
  sudo su
  ```

2. Enter the password of the rescue user, displayed in the Dedibox console, and confirm by pressing enter. 

  <Message type="tip">

  For security reasons, the password will not be shown when you type it.

  </Message>

## How to mount the partitions

> To have access to your data using the rescue moude, you have to mount your partitions.

Reviewer 2: ```suggestion
To get access to your data in rescue mode, you have to mount your partitions.
```

If this step succeeds, then you are almost sure to be able to recover your files.

Mount your partitions by entering, as `root` user, the command `mountall.sh`.  
The partitions are then mounted in the `/mnt` directory

<Message type="note">

> Some errors such as `the swap cannot be mounted` or `many partitions do not exist`. You can safely ignore these errors. 

Reviewer 2: ```suggestion
Some errors such as `the swap cannot be mounted` or `many partitions do not exist` may display. You can safely ignore these errors. 
```

</Message> 

## How to recover your data

> You can recover your data either by transferring them to the [Dedibackup](https://www.scaleway.com/en/dedibox/storage/dedibackup/) storage space or by downloading it directly to your local computer using the SFTP protocol.

Reviewer 2: ```suggestion
You can recover your data either by transferring it to the [Dedibackup](https://www.scaleway.com/en/dedibox/storage/dedibackup/) storage space or by downloading it directly to your local computer using the SFTP protocol.
```

<Message type="important">

> Be careful when editing your data, the `root` user has full permissions on each file and can perform actions like irreversible deletion of data.

Reviewer 2: ```suggestion
Be careful when editing your data. The `root` user has full permissions on each file and as such can perform actions which could be undesirable, including the irreversible deletion of data.
```

</Message>

###  How to resuce your data using Dedibackup

The following procedure shows how to retrieve all the contents of your disk (including all partitions) on your [Dedibackup](https://www.scaleway.com/en/dedibox/storage/dedibackup/) storage space.

1. Install an FTP client and `screen` via the following command. `screen` allows the FTP client to run without your computer having to stay on with Putty open.

  ```
  apt -y install screen lftp
  ```

2. Launch `screen` bt typing the following command. Then press enter to launch the application:

  ```
  screen
  ```

3. Lauch the transfer of your data by running the following command:

  ```
  lftp -u YOUR_DEDIBACKUP_LOGIN,YOUR_DEDIBACKUP_PASSWORD dedibackup.online.net -e "mirror --reverse /mnt /; exit"
  ```

  * `YOUR_DEDIBACKUP_LOGIN` has to be replaced with your dedibackup login (typically sd- your server number).
  * `YOUR_DEDIBACKUP_PASSWORD` has to be replaced with your dedibackup password.

  <Message type="tip">

  If you do not know the password, change it in the console to get a new password.

  </Message>

> You should now be able to browse and retrieve all your files with any ftp client by connecting to dedibackup via an FTP client.

Reviewer 2: ```suggestion
You should now be able to browse and retrieve all your files by connecting to Dedibackup with any FTP client.
```

<Message type="tip">

The Dedibackup service is limited to 1000 files per volume. It is recommended to use archives when backing up your data.

</Message>

### How to back up to your personal computer (via SFTP)

The recovery mode supports connection using SSH by default. It is therefore possible to connect from your computer using a SFTP client to recover your files.

> In this example we use the client [FileZilla](en/web/web-hosting/tutorials/ftp/configure-filezilla)

Reviewer 2: broken link
